lib/fa2/asset/multi_asset.impl.jsligo

```diff
@@ -0,0 +1,263 @@
+            assert_token_exist(s, token_id);
+            assert_authorisation([s.operators, from_, token_id]);
+            let ledger =
+               decrease_token_amount_for_user([l, from_, token_id, amount]);
+            ledger
+            = increase_token_amount_for_user([ledger, to_, token_id, amount]);
+            return ledger
+         };
+      const process_single_transfer = ([l, t]: [Ledger, TZIP12.transfer_from]): Ledger => {
+         const { from_, txs } = t;
+         const ledger = List.fold_left(process_atomic_transfer(from_), l, txs);
+         return ledger
+      };
+      const ledger = List.fold_left(process_single_transfer, s.ledger, t);
+      //refresh set of owner,token_ids
+
+      return [list([]), set_ledger([s, ledger])]
+   };
+   // Balance_of entrypoint
+
+   @entry
+   const balance_of = (b: TZIP12.balance_of, s: storage): [
+      list<operation>,
+      storage
+   ] => {
+      const { requests, callback } = b;
+      const get_balance_info = (request: TZIP12.request): TZIP12.callback => {
+         const { owner, token_id } = request;
+         assert_token_exist(s, token_id);
+         const balance_ = get_for_user([s.ledger, owner, token_id]);
+         return ({ request: request, balance: balance_ })
+      };
+      const callback_param = List.map(get_balance_info, requests);
+      const operation =
+         Tezos.transaction(Main(callback_param), 0mutez, callback);
+      return [list([operation]), s]
+   };
+   // Update_operators entrypoint
+
+   @entry
+   const update_operators = (updates: TZIP12.update_operators, s: storage): [
+      list<operation>,
+      storage
+   ] => {
+      const update_operator = (
+         [operators, update]: [Operators, TZIP12.unit_update]
+      ): Operators =>
+         match(update) {
+            when (Add_operator(operator)):
+               add_operator(
+                  [
+                     operators,
+                     operator.owner,
+                     operator.operator,
+                     operator.token_id
+                  ]
+               )
+            when (Remove_operator(operator)):
+               remove_operator(
+                  [
+                     operators,
+                     operator.owner,
+                     operator.operator,
+                     operator.token_id
+                  ]
+               )
+         };
+      let operators = get_operators(s);
+      operators = List.fold_left(update_operator, operators, updates);
+      const store = set_operators([s, operators]);
+      return [list([]), store]
+   };
+   @view
+   const get_balance = (p: [address, nat], s: storage): nat => {
+      const [owner, token_id] = p;
+      assert_token_exist(s, token_id);
+      return match(Big_map.find_opt([owner, token_id], s.ledger)) {
+         when (None()):
+            0n
+         when (Some(n)):
+            n
+      }
+   };
+   @view
+   const total_supply = (token_id: nat, s: storage): nat =>
```

Comment:
Is it a choice to keep `s` instead of `_s `? Currently there is warning. same for token_id and other functions

Reply:
I removed warnings